src/enrichment_tables/pgtable.rs

```diff
@@ -0,0 +1,440 @@
+    // #[snafu(display("host missing"))]
+    // HostMissing,
+    // #[snafu(display("multiple hosts not supported: {:?}", hosts))]
+    // MultipleHostsNotSupported { hosts: Vec<Host> },
+}
+
+/// Configuration of TLS when connecting to PostgreSQL.
+#[configurable_component]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+struct PgtableTlsConfig {
+    /// Absolute path to an additional CA certificate file.
+    ///
+    /// The certificate must be in the DER or PEM (X.509) format.
+    #[configurable(metadata(docs::examples = "certs/ca.pem"))]
+    ca_file: PathBuf,
+}
+
+/// Postgres-specific settings
+#[configurable_component(enrichment_table("pgtable"))]
+#[derive(Clone, Debug, Default)]
+pub struct PgtableConfig {
+    #[configurable(derived)]
+    /// The postgres connection string
+    endpoint: String,
+    /// Table to query
+    table: String,
+    /// Columns to project from table
+    columns: Vec<String>,
+    /// TLS client certificate
+    #[configurable(derived)]
+    tls: Option<PgtableTlsConfig>,
+}
+impl_generate_config_from_default!(PgtableConfig);
+
+fn surround_with_quotes(name: String) -> String {
+  return ["\"", &name, "\""].concat();
+}
+
+fn row_to_vec_value(row: Row) -> Vec<Value> {
+    let row_len = row.len();
+    let mut dst: Vec<Value> = Vec::new();
+    for i in 0..row_len {
+      let rint : Result<i64, PgError> = row.try_get(i);
+      match rint {
+        Err(_) => {
+          let rstr : Result<String, PgError> = row.try_get(i);
+          match rstr {
+            Ok(s) => {
+              dst.push(Value::Bytes(Bytes::from(s)));
+            }
+            Err(_) => {
+              dst.push(Value::Null);
+            }
+          }
+        }
+        Ok(i) => {
+          dst.push(Value::Integer(i));
+        }
+      }
+    };
+    return dst;
+}
+
+impl EnrichmentTableConfig for PgtableConfig {
+    async fn build(
+        &self,
+        _globals: &crate::config::GlobalOptions,
+    ) -> crate::Result<Box<dyn Table + Send + Sync>> {
+        let config: Config = self.endpoint.parse()?;
+        let client = match &self.tls {
+            Some(tls_config) => {
+                let mut builder = SslConnector::builder(SslMethod::tls_client())?;
+                builder.set_ca_file(tls_config.ca_file.clone())?;
+                let connector = MakeTlsConnector::new(builder.build());
+
+                let (client, connection) = config.connect(connector).await?;
+                tokio::spawn(connection);
+                client
+            }
+            None => {
+                let (client, connection) = config.connect(NoTls).await?;
+                tokio::spawn(connection);
+                client
+            }
+        };
+        let columns_with_commas : Vec<String> = intersperse(self.columns.clone().into_iter().map(surround_with_quotes), ",".to_string()).collect();
+        let select_query : String = ["SELECT ", &columns_with_commas.concat(), " FROM ", &self.table].concat();
```

Comment:
It would be more idiomatic Rust to use `format!`:

```suggestion
        let select_query = format!("SELECT {} FROM {}", &columns_with_commas.concat(), &self.table);
```